README.adoc

```diff
@@ -33,81 +41,100 @@ Static site generation based on {url-antora}[Antora] using {url-asciidoctor}:[As
 While you can install Node from the official packages, we strongly recommend that you use {url-nvm}[nvm] (Node Version Manager) to install and manage Node.
 Follow the {url-nvm-install}[nvm installation instructions] to set up nvm on your machine.
 
-NOTE: all npm commands are intended to be run from the repository root folder.
+[NOTE]
+====
+* all commands are intended to be run from the repository root folder.
+* the generated documentation is available in the `build/site` folder.
+====
 
-=== Build the static site for production
+Don't forget to run `npm install` the first time you build the project or on `package.json` changes.
 
-Run
-```bash
-npm install
-npm run build
-```
-The static site is then available in the `build/site` folder.
+Check that the `Antora` CLI is available by calling `./node_modules/.bin/antora --version`.
 
 
-=== Local development
+[#non-production-build]
+== Non production build
 
-The `antora` CLI is available by calling `./node_modules/.bin/antora --version`
+[[preview]]
+=== Preview support
 
-For more CLI options, see https://docs.antora.org/antora/2.3/cli/options/
+The easiest way is to use a script that generates the documentation partially and/or with specific settings for both local
+developments, Pull Request previews or to produce documentation archives.
 
+This includes building the documentation using:
 
-==== Building for local developments
+* already checkout local documentation content repositories
+* single branch for all components
+* single branch of a single component
+* a set of branches of various components. This is let validate https://docs.antora.org/antora/2.3/page/version-and-component-xrefs/[references between pages that belong to different docs components and/or versions]
+* ....
 
-[WARNING]
-====
-The following command https://docs.antora.org/antora/2.3/playbook/runtime-fetch/[fetch the documentation content repositories] each time they are run. +
-<<local-content>> or <<preview>> can prevent this to happen.
-====
+To know all available options, run
+[source,shell script]
+----
+./build-preview.bash --help
+----
+
```

Comment:
I would put around here an example of the command line (copy past ready) to build and preview the site for a given branch with local sources -> It's an interesting use case IMHO, it allows contributors to have a preview without opening a pull request, which is what you wanna do when you are working on a big contribution

Reply:
yes good idea, I will also put the example to build cross component versions like in https://github.com/bonitasoft/bonita-documentation-site/pull/222#issuecomment-814609850

Reply:
done in 4227066, feel free to update the example and explanation if it is uncleared. As I have developed the script and use it a lot, I am biased and don't know really what is needed to let a new comer understand how it works 😄